kcp/setup-kcp-cps.sh

```diff
@@ -0,0 +1,169 @@
+
+create_kubeconfig_secret() {
+    sa_name=$1
+    sa_secret_name=$(KUBECONFIG="${CPS_KUBECONFIG}" kubectl get sa $sa_name -n $ARGOCD_NAMESPACE -o=jsonpath='{.secrets[0].name}')
+
+    ca=$(KUBECONFIG="${CPS_KUBECONFIG}" kubectl get secret/$sa_secret_name -n $ARGOCD_NAMESPACE -o jsonpath='{.data.ca\.crt}')
+    token=$(KUBECONFIG="${CPS_KUBECONFIG}" kubectl get secret/$sa_secret_name -n $ARGOCD_NAMESPACE -o jsonpath='{.data.token}' | base64 --decode)
+    namespace=$(KUBECONFIG="${CPS_KUBECONFIG}" kubectl get secret/$sa_secret_name -n $ARGOCD_NAMESPACE -o jsonpath='{.data.namespace}' | base64 --decode)
+
+    server=$(KUBECONFIG="${CPS_KUBECONFIG}" kubectl config view -o jsonpath='{.clusters[?(@.name == "workspace.kcp.dev/current")].cluster.server}')
+
+    secret_name=$2
+    kubeconfig_secret="
+apiVersion: v1
+kind: Secret
+metadata:
+  creationTimestamp: null
+  name: ${secret_name}
+  namespace: ${ARGOCD_NAMESPACE}
+stringData:
+  kubeconfig: |
+    apiVersion: v1
+    kind: Config
+    clusters:
+    - name: default-cluster
+      cluster:
+        certificate-authority-data: ${ca}
+        server: ${server}
+    contexts:
+    - name: default-context
+      context:
+        cluster: default-cluster
+        namespace: ${ARGOCD_NAMESPACE}
+        user: default-user
+    current-context: default-context
+    users:
+    - name: default-user
+      user:
+        token: ${token}
+"
+
+    echo "${kubeconfig_secret}" | KUBECONFIG="${CPS_KUBECONFIG}" kubectl apply -f -
+}
+
+echo "Installing Argo CD resources in workspace $WORKSPACE and namespace $ARGOCD_NAMESPACE"
+KUBECONFIG="${CPS_KUBECONFIG}" kubectl create ns $ARGOCD_NAMESPACE || true
+
+KUBECONFIG="${CPS_KUBECONFIG}" kubectl apply -f $ARGOCD_MANIFEST -n $ARGOCD_NAMESPACE
+
+echo "Creating KUBECONFIG secrets for argocd server and argocd application controller service accounts"
+create_kubeconfig_secret "argocd-server" "kcp-kubeconfig-controller"
+create_kubeconfig_secret "argocd-application-controller" "kcp-kubeconfig-server"
+
+echo "Verifying if argocd components are up and running after mounting kubeconfig secrets"
+KUBECONFIG="${CPS_KUBECONFIG}" kubectl wait --for=condition=available deployments/argocd-server -n $ARGOCD_NAMESPACE
+count=0
+while [ $count -lt 30 ]
+do
+    count=`expr $count + 1`
+    replicas=$(KUBECONFIG="${CPS_KUBECONFIG}" kubectl get statefulsets argocd-application-controller -n $ARGOCD_NAMESPACE -o jsonpath='{.spec.replicas}')
+    ready_replicas=$(KUBECONFIG="${CPS_KUBECONFIG}" kubectl get statefulsets argocd-application-controller -n $ARGOCD_NAMESPACE -o jsonpath='{.status.readyReplicas}')
+    if [ "$replicas" -eq "$ready_replicas" ]; then 
+        break
+    fi
+    if [ $count -eq 30 ]; then 
+        echo "Statefulset argocd-application-controller does not have the required replicas"
+        exit 1
+    fi
+done
+
+cat <<EOF | kubectl apply -n $ARGOCD_NAMESPACE -f -
```

Comment:
This part of the kubectl should use the CPS kubeconfig right and not the cluster kubeconfig?

Reply:
Good catch! I did not face any issues because the KUBECONFIG variable already existed.